src/sentry/stacktraces.py

```diff
@@ -0,0 +1,201 @@
+    together with some meta information.
+    """
+    rv = []
+
+    def _report_stack(stacktrace, container):
+        platforms = set()
+        for frame in stacktrace.get('frames') or ():
+            platforms.add(frame.get('platform') or data['platform'])
+        rv.append(StacktraceInfo(
+            stacktrace=stacktrace,
+            container=container,
+            platforms=platforms
+        ))
+
+    exc_container = data.get('sentry.interfaces.Exception')
+    if exc_container:
+        for exc in exc_container['values']:
+            stacktrace = exc.get('stacktrace')
+            if stacktrace:
+                _report_stack(stacktrace, exc)
+
+    stacktrace = data.get('sentry.interfaces.Stacktrace')
+    if stacktrace:
+        _report_stack(stacktrace, None)
+
+    threads = data.get('threads')
+    if threads:
+        for thread in threads['values']:
+            stacktrace = thread.get('stacktrace')
+            if stacktrace:
+                _report_stack(stacktrace, thread)
+
+    return rv
+
+
+def should_process_for_stacktraces(data):
+    from sentry.plugins import plugins
+    infos = find_stacktraces_in_data(data)
+    platforms = set()
+    for info in infos:
+        platforms.update(info.platforms or ())
+    for plugin in plugins.all(version=2):
+        processors = safe_execute(plugin.get_stacktrace_processors,
+                                  data=data, stacktrace_infos=infos,
+                                  platforms=platforms,
+                                  _with_transaction=False)
+        if processors:
+            return True
+    return False
+
+
+def get_processors_for_stacktraces(data, infos):
+    from sentry.plugins import plugins
+
+    platforms = set()
+    for info in infos:
+        platforms.update(info.platforms or ())
+
+    processors = []
+    for plugin in plugins.all(version=2):
+        processors.extend(safe_execute(plugin.get_stacktrace_processors,
+                                       data=data, stacktrace_infos=infos,
+                                       platforms=platforms,
+                                       _with_transaction=False) or ())
+
+    if processors:
+        project = Project.objects.get_from_cache(id=data['project'])
+        processors = [x(data, infos, project) for x in processors]
+
+    return processors
+
+
+def process_single_stacktrace(stacktrace_info, processors):
+    # TODO: associate errors with the frames and processing issues
+    changed_raw = False
+    changed_processed = False
+    raw_frames = []
+    processed_frames = []
+    all_errors = []
+
+    for frame in stacktrace_info.stacktrace['frames']:
+        need_processed_frame = True
+        need_raw_frame = True
+        errors = None
+        for processor in processors:
+            try:
+                rv = processor.process_frame(frame)
+                if rv is None:
+                    continue
+            except Exception:
```

Comment:
The intention is that all processors get a choice to process the frame. The first one that manages is the winning one. Processors need to only handle platforms they know about. Not sure if we can improve on this to make it less error prone if a processor forgets to check for platform.